dockerfile-image-update/src/test/java/com/salesforce/dockerfileimageupdate/model/FromInstructionTest.java

```diff
@@ -0,0 +1,224 @@
+package com.salesforce.dockerfileimageupdate.model;
+
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+public class FromInstructionTest {
+
+    @DataProvider
+    public Object[][] inputFormInstructionData() {
+        return new Object[][]{
+                {"FROM dockerimage:3 # some comment", "FROM dockerimage:3 # some comment"},
+                {"FROM dockerimage:3 AS test", "FROM dockerimage:3 AS test"},
+                {"FROM dockerimage:3 \tAS \ttest # some comment", "FROM dockerimage:3 AS test # some comment"},
+                {"FROM    dockerimage:3#   some   comment", "FROM dockerimage:3 #   some   comment"},
+                {"        FROM       dockerimage   ", "FROM dockerimage"},
+                {"\t FROM \t dockerimage:4 \t #comment", "FROM dockerimage:4 #comment"},
+                {"FROM dockerimage:4:4:4 #comment", "FROM dockerimage:4 #comment"},
+                {"FROM dockerimage:4 #comment me # # ", "FROM dockerimage:4 #comment me # # "}
+        };
+    }
+
+    @Test(dataProvider = "inputFormInstructionData")
+    public void testStringResult(String fromInstruction, String expectedResult) {
+        assertEquals(new FromInstruction(fromInstruction).toString(), expectedResult);
+    }
+
+    @DataProvider
+    public Object[][] isFromInstructionData() {
```

Comment:
how about `#FROM dockerimage:3` and `# FROM dockerimage:3` and ` # FROM dockerimage:3`

Reply:
Good call @npwolf. I just added those cases to this and also added a case to the invalid data array in 5f078b5. Fresh eyes told me to reformat these "tables" also.